src/core/hle/service/nvdrv/devices/nvhost_ctrl.cpp

```diff
@@ -155,7 +155,14 @@ NvResult nvhost_ctrl::IocCtrlEventRegister(const std::vector<u8>& input, std::ve
         return NvResult::BadParameter;
     }
     if (events_interface.registered[event_id]) {
-        return NvResult::BadParameter;
+        const auto event_state = events_interface.status[event_id];
+        if (event_state == EventState::Registered || event_state == EventState::Waiting ||
```

Comment:
```suggestion
        if (event_state != EventState::Free)
```

Would simplify the check a bit more.